### ISSUE TITLE
Modifying Torch pretrained models

### ISSUE THREAD

**Issue by mescarra** [NVIDIA/DIGITS#1748]
Hi!

I'm training Torch DCGAN model (without digits) with unsupervised data. Then I generate a pretrained torch model with DCGAN's discriminator and only change the last layer to have the same amount of maps as labels in my supervised dataset. The model alone (without the weights) works just fine, but when I add the weights I get the error below. I wonder if that is because I modified that one layer, and in that case I would like to know how to solve it (with Caffe models you just need to change that layer's name and it won't try to load weights for it).

Thanks in advance.

```
2017-07-26 11:18:53 [20170726-111849-26d6] [ERROR] Train Torch Model: /usr/share/lua/5.1/optim/ConfusionMatrix.lua:117: bad argument #1 to 'indexAdd' (out of range at /tmp/buildd/torch7-0.9.99/pkg/torch/lib/TH/generic/THTensor.c:729)
2017-07-26 11:18:53 [20170726-111849-26d6] [WARNING] Train Torch Model unrecognized output: DIGITS Lua Error
2017-07-26 11:18:53 [20170726-111849-26d6] [WARNING] Train Torch Model unrecognized output: stack traceback:
2017-07-26 11:18:53 [20170726-111849-26d6] [WARNING] Train Torch Model unrecognized output: [C]: in function 'indexAdd'
2017-07-26 11:18:53 [20170726-111849-26d6] [WARNING] Train Torch Model unrecognized output: /usr/share/lua/5.1/optim/ConfusionMatrix.lua:117: in function 'batchAdd'
2017-07-26 11:18:53 [20170726-111849-26d6] [WARNING] Train Torch Model unrecognized output: .../lib/python2.7/dist-packages/digits/tools/torch/main.lua:701: in function 'Validation'
2017-07-26 11:18:53 [20170726-111849-26d6] [WARNING] Train Torch Model unrecognized output: .../lib/python2.7/dist-packages/digits/tools/torch/main.lua:840: in main chunk
2017-07-26 11:18:53 [20170726-111849-26d6] [WARNING] Train Torch Model unrecognized output: [C]: in function 'dofile'
2017-07-26 11:18:53 [20170726-111849-26d6] [WARNING] Train Torch Model unrecognized output: ...b/python2.7/dist-packages/digits/tools/torch/wrapper.lua:25: in function <...b/python2.7/dist-packages/digits/tools/torch/wrapper.lua:25>
2017-07-26 11:18:53 [20170726-111849-26d6] [WARNING] Train Torch Model unrecognized output: [C]: in function 'xpcall'
2017-07-26 11:18:53 [20170726-111849-26d6] [WARNING] Train Torch Model unrecognized output: ...b/python2.7/dist-packages/digits/tools/torch/wrapper.lua:25: in main chunk
2017-07-26 11:18:53 [20170726-111849-26d6] [WARNING] Train Torch Model unrecognized output: [C]: in function 'dofile'
2017-07-26 11:18:53 [20170726-111849-26d6] [WARNING] Train Torch Model unrecognized output: /usr/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
2017-07-26 11:18:53 [20170726-111849-26d6] [WARNING] Train Torch Model unrecognized output: [C]: at 0x00406670
2017-07-26 11:18:53 [20170726-111849-26d6] [ERROR] Train Torch Model task failed with error code 1
```

**Comment by ethantang95** [comment]
This example shows how fine tuning and weight changing works in Torch. https://github.com/NVIDIA/DIGITS/tree/master/examples/fine-tuning#using-torch7. Hopefully this can help.

**Comment by mescarra** [comment]
Thank you very much Ethan. I had to `net:remove` some top layers and `net:add` custom ones in the `fineTuneHook` function, and it works just fine.
